status/templatetags/status_filters.py

```diff
@@ -122,7 +123,10 @@ def show_detail(submission, user):
     if contests:
         contests = contests.filter(problem=submission.problem)
         for contest in contests:
-            if user == contest.owner or user in contest.coowner.all():
+            contestants = Contestant.objects.filter(contest=contest)
```

Comment:
In the for loop, I think the order of checking can be slightly modified:
```
if not (user == contest.owner or user in contest.coowner.all())
    return False
contestants = ...
if submission.user in contestants:
    return True
```
BTW, this section of code is duplicated, can you function-lize it?